webhooks-extension/src/components/WebhookDisplayTable/tests/WebhookTable.test.js

```diff
@@ -85,4 +85,16 @@ describe('with webhooks', () => {
     const { getByText } = renderWithRouter(<WebhookDisplayTable match={{}} selectedNamespace="default"/>);
     await waitForElement(() => getByText("No webhooks created under namespace 'default', click 'Add Webhook' button to add a new one."));
   });
+
+  it('display branch modal when table row clicked', async () => {
+    jest.spyOn(API, "getWebhooks").mockImplementation(() => Promise.resolve(webhooks));
+    jest.spyOn(API, "fetchBranches").mockImplementation(() => Promise.resolve([]));
+    const { getByTestId } = renderWithRouter(<WebhookDisplayTable match={{}} selectedNamespace="*"/>);
+
+    await waitForElement(() => getByTestId(/webhook-table-row-first-test-webhook/i));
+
+    fireEvent.click(getByTestId(/webhook-table-row-first-test-webhook/i));
```

Comment:
we should only use data-testid if there's no user-facing content we can reference in the tests, e.g. button text, headers, etc.